### PR TITLE
NAS-118360 / 22.12 / Handle clipboard API not available state

### PR DESCRIPTION
**Pull request by AlexKarpov98**
read comments below ⬇️

### CODE REVIEW

**Comment by denysbutenko** [comment]
Actually I think it could be wrong usage of execCommand() after my recent changes, needs to check git history if goal to make it works

**Comment by AlexKarpov98** [comment]
> Actually I think it could be wrong usage of execCommand() after my recent changes, needs to check git history if goal to make it works

Screen Shot 2022-10-04 at 16 22 31

@denysbutenko anyways it's deprecated and will be removed soon.
https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand

**Comment by AlexKarpov98** [comment]
@denysbutenko @undsoft 

I see that clipboard is blocked
http://m40g3-137.dc1.ixsystems.net/ui/dashboard
and
http://10.234.30.8/ui/dashboard

☝️ **navigator clipboard API needs a secure context (https)** 

made an update with document.execCommand(‘copy’) and textarea, it's a working solution!

File: src/app/core/components/copy-btn/copy-btn.component.ts [hunk 1]
```diff
@@ -17,15 +15,31 @@ export class CopyButtonComponent {
...
+  private showSuccessMessage(): void {
+    this.snackbar.success(this.translate.instant('Copied to clipboard'));
+  }
+
+  private handleCopyToClipboard(textToCopy: string): Promise<void> {
+    if (navigator.clipboard) {
+      return navigator.clipboard.writeText(textToCopy);
     }
+
+    const textArea = document.createElement('textarea');
```
Comment (undsoft) [inline src/app/core/components/copy-btn/copy-btn.component.ts:27]: Worth extracting to a separate method.

File: src/app/core/components/copy-btn/copy-btn.component.ts [hunk 2]
```diff
@@ -17,15 +15,31 @@ export class CopyButtonComponent {
...
+    textArea.value = textToCopy;
+    textArea.style.position = 'fixed';
+    textArea.style.left = '-9999px';
+    textArea.style.top = '-9999px';
+    document.body.appendChild(textArea);
+    textArea.focus();
+    textArea.select();
+    return new Promise((resolve, reject) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
```
Comment (undsoft) [inline src/app/core/components/copy-btn/copy-btn.component.ts:36]: You can do:
```
if (condition) {
  statement();
} else {
  statement();
}
```
and not disable linter here.

**Review by undsoft — APPROVED** [review]
Tested on https. Works well.

**Pull request merged**